app/src/marketplace/components/ExpirationCounter/index.jsx

```diff
@@ -0,0 +1,59 @@
+// @flow
+
+import React, { useState } from 'react'
+import cx from 'classnames'
+
+import useInterval from '$shared/hooks/useInterval'
+
+import styles from './expirationCounter.pcss'
+
+type Props = {
+    expiresAt: Date,
+    className?: string,
+}
+
+const DAY_SECONDS = 24 * 60 * 60
+
+const getExpirationString = (secondsLeft: number): string => {
+    if (secondsLeft < DAY_SECONDS) {
+        const date = new Date(0)
+        date.setTime(secondsLeft * 1000)
+        return date.toISOString().substr(11, 8) // hh:mm:ss
+    }
+
+    const days = parseInt(secondsLeft / DAY_SECONDS, 10)
+    return `${days} day${days > 1 ? 's' : ''}`
+}
+
+const Error = ({ expiresAt, className }: Props) => {
+    const [secondsUntilExpiration, setSecondsUntilExpiration] = useState(parseInt((expiresAt.getTime() - Date.now()) / 1000, 10))
+
+    useInterval(() => {
+        setSecondsUntilExpiration(secondsUntilExpiration - 1)
```

Comment:
Ahh, actually, this is no good. `setInterval`/`setTimeout` is not guaranteed to actually fire on the interval you provide. See "Reasons for delays longer than specified" in https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Notes

e.g. just now I ran `console.time('timeout'); setTimeout(() => {console.timeEnd('timeout');}, 1000)`  in devtools:
![image](https://user-images.githubusercontent.com/43438/66802268-7622f200-ef4e-11e9-9d6a-eabce1f367ec.png)

So as-is, this timer will get increasingly inaccurate the longer it's displayed on the page.

Easy fix is to always do the full calculation i.e  `parseInt((expiresAt.getTime() - Date.now()) / 1000, 10)` instead of `secondsUntilExpiration - 1`